modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientTupleSerializer.java

```diff
@@ -456,14 +456,11 @@ public static PartitionAwarenessProvider getPartitionAwarenessProvider(Collectio
                 return hash;
             }
 
-            while (iter.hasNext()) {
-                Tuple next = iter.next();
-                if (hash != getColocationHash(schema, next)) {
-                    return null;
-                }
+            // Temporary disable direct mapping for batches.
```

Comment:
Do we have a ticket to remove temporary behavior?

Reply:
Created a ticket [1], added TODO.
[1] https://issues.apache.org/jira/browse/IGNITE-25348